repos/system_upgrade/common/actors/peseventsscanner/libraries/peseventsscanner_repomap.py

```diff
@@ -0,0 +1,296 @@
+from leapp.libraries.common.config.version import get_source_major_version, get_target_major_version
+from leapp.libraries.stdlib import api
+
+
+DEFAULT_PESID = {
+    '7': 'rhel7-base',
+    '8': 'rhel8-BaseOS',
+    '9': 'rhel9-BaseOS'
+}
+
+
+def _get_channel_prio(pesid_repo):
+    priorities = {
+        'beta': 0,
+        'ga': 1,
+    }
+    return priorities.get(pesid_repo.channel, 10)
+
+
+class RepoMapDataHandler(object):
+    """
+    Provide the basic functionality to work with the repository data easily.
+    """
+
+    def __init__(self, repo_map, default_channels=None, prio_channel=None):
+        """
+        Initialize the object based on the given RepositoriesMapping msg.
+
+        Expects that msg contains just stuff related for the current IPU
+        (at least mapping and repos for the used upgrade path and architecture).
+
+        :param repo_map: A valid RepositoryMapping message.
+        :type repo_map: RepositoryMapping
+        :param default_channels: A list of default channels to use when a target repository
+                                 equivalent exactly matching a source repository was not found.
+        :type default_channels: List[str]
+        :param prio_channel: Prefer repositories with this channel when looking for target equivalents.
+        :type prio_channel: str
+        """
+        # NOTE: currently I am keeping this default data structure that is not
```

Comment:
Could you please be more specific about why it is not ideal? I know it is a comment but it can help someone else later.

Reply:
Basically if we deserialize the data in a little bit different data stracture, we could be more effective. e.g. why to search for particular repositories with a specific pesid when you could have just dict {pesid: [repos]}. etc. but keeping it as it is for now as it doesn't matter so much as we should manipulate with the data mainly via defined methods.